### PR TITLE
Added timeout option for OLED boot logo.

### PR DESCRIPTION
**Pull request by luttermann**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/doio/kb16/rev1/config.h [hunk 1]
```diff
@@ -20,6 +20,9 @@
 /* Use the custom font */
 #define OLED_FONT_H "./lib/glcdfont.c"
 
+/* Set timeout for displaying boot logo */
+#define OLED_LOGO_TIMEOUT 5000
+
```
Comment (zvecr) [inline keyboards/doio/kb16/rev1/config.h:25]:
```suggestion
```

File: keyboards/doio/kb16/rev2/config.h [hunk 2]
```diff
@@ -23,6 +23,9 @@
     #define I2C1_SDA_PIN B11
     #define I2C_DRIVER I2CD2
 
+    /* Set timeout for displaying boot logo */
+    #define OLED_LOGO_TIMEOUT 5000
+
```
Comment (zvecr) [inline keyboards/doio/kb16/rev2/config.h:28]:
```suggestion
```

File: keyboards/doio/kb16/kb16.c
Comment (zvecr) [inline keyboards/doio/kb16/kb16.c:1]: Somewhere at the top of the file,
```c
/* Default timeout for displaying boot logo */
#ifndef OLED_LOGO_TIMEOUT
    #define OLED_LOGO_TIMEOUT 5000
#endif
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/doio/kb16/kb16.c
Comment (luttermann) [reply keyboards/doio/kb16/kb16.c:1]: You're right @zvecr, it doesn't make sense to have the configuration separated for `rev1` and `rev2`, but wouldn't it make more sense to put it into a new header file `keyboards/doio/kb16/config.h` instead?

That could be done for all the common settings for the two revisions (Thats most of them, I'll gladly make a separate PR for that, or update this one).

Comment (zvecr) [reply keyboards/doio/kb16/kb16.c:1]: This particular setting does not need to be in a config.h at all. Doing so then requires users to undef when specifying their override.

Comment (luttermann) [reply keyboards/doio/kb16/kb16.c:1]: Thanks for the information and suggestions :smiley:

**Review by drashna — APPROVED** [review]

**Review by zvecr — APPROVED** [review]

**Pull request merged**